✨ (INT-1873) 3PB Documentation - Add e-bot7 Connector (#1670)
LivePersonInc/developers-community: pages/Documents/ConversationalAI/Third-PartyBots/e-bot7/basic-content.md

---
pagename: Basic Content
sitesection: Documents
categoryname: "Conversational AI"
documentname: Third-Party Bots
subfoldername: e-bot7
permalink: third-party-bots-e-bot7-basic-content.html
indicator:
---

### The Welcome Event

> The behaviour of the welcome event is different depending on whether the bot is for chat and messaging. This divergence comes down to the way that each individual Liveperson product works.

Reviewer: ```suggestion
The behavior of the welcome event is different depending on whether the bot is for chat and messaging. This divergence comes down to the way that each individual Liveperson product works.
```

> The behaviour of the welcome event is different depending on whether the bot is for chat and messaging. This divergence comes down to the way that each individual Liveperson product works.

Reviewer 2: ```suggestion
The behaviour of the welcome event is different depending on whether the bot is for chat and messaging. This divergence comes down to the way that each individual LivePerson product works.
```

#### Messaging Conversations

A Messaging conversation qualifies as "initiated" from a Conversational Cloud perspective only after the consumer sends
their first message. The consumer is prompted for their initial message in the channel they have chosen to initiate the

> conversation. As a result, the consumer’s first message will be the entry point for the main flow in e-bot7 which can be processed by a `Visitor Input` node.

Reviewer 2: ```suggestion
conversation. As a result, the consumer’s first message will be the entry point for the main flow in e-bot7, which can be processed by a `Visitor Input` node.
```

In this scenario, all bot messages in the flow which are created before this node will be skipped.

<img class="fancyimage" style="width:450px" src="img/ThirdPartyBots/e-bot7-messaging-flow.png">
<figure>Figure 2.1 Example conversation flow for messaging without support of the `Welcome` event</figure>
<br>

#### Chat Conversations

> A Chat conversation is considered started when the chat is routed to an agent. Best practice is for the agent to provide the first response.

Reviewer: ```suggestion
A Chat conversation is considered started when the chat is routed to an agent. The best practice is for the agent to provide the first response.
```
```suggestion
A Chat conversation is considered started when the chat is routed to an agent. The best practice is for the agent to provide the first response.
```

In this scenario, all bot messages in the flow which are created before the first `Visitor Input` node will be processed.

> Make sure to have `Bot Message` nodes after the `Start Flow` entrypoint to utilise the `Welcome` event.

Reviewer: ```suggestion
Make sure to have `Bot Message` nodes after the `Start Flow` entry point to utilize the `Welcome` event.
```

<img class="fancyimage" style="width:450px" src="img/ThirdPartyBots/e-bot7-chat-flow.png">
<figure>Figure 2.2 Example conversation flow for chats with support of the `Welcome` event</figure>
<br>

### Bot Messages

> The flow of a bot in e-bot7 can include three different types of messages a bot can send. These are `Bot Message: Text`, `Bot Message: Image`, `Bot Message: Video`. All three types are supported with some restrictions:

Reviewer: Maybe add all of these types in the table and make a column `Restrictions` that contains  if any restriction applies to the given type

- `Bot Message: Text` is rendered as pure text and contains no markup.

### Visitor Input

> Input messages by a visitor which will be delivered to a bot are of different types. Currently the types `Visitor Input: Text` and `Visitor Input: Selection` are supported.

Reviewer: maybe something similar to the above,  make it into the table form

> Input messages by a visitor which will be delivered to a bot are of different types. Currently the types `Visitor Input: Text` and `Visitor Input: Selection` are supported.

Reviewer 2: ```suggestion
Input messages by a visitor which will be delivered to a bot can have different types. Currently the types `Visitor Input: Text` and `Visitor Input: Selection` are supported.
```

### Bot Actions

Bots can trigger different forms of actions.

{: .important}

> Please note we only support **ONE ACTION** per response

Reviewer 2: ```suggestion
Please note we only support **ONE ACTION** per response.
```

#### Transfer

If the bot needs to transfer the conversation to a human agent, or the conversation flow indicates that another bot is better suited for the identified intent, you will need to tell the connector to transfer the conversation to a given skill.

This is achieved using a special `Bot Message: Text`.

Multiple scenarios for transfer/escalations exist triggered by the transfer action object.

> 1. Explicit request from visitor to transfer to an agent (Eg, action : transfer)

Reviewer: ```suggestion
1. Explicit request from a visitor to transfer to an agent (Eg, action: transfer)
```

> 2. If the Bot does not have an appropriate answer, it should recognise this as a scenario for a transfer.

Reviewer: ```suggestion
2. If the Bot does not have an appropriate answer, it should recognize this as a scenario for a transfer.
```

>    Depending on the connector configuration or the decision making capacity of the bot, the bot will transfer to a particular agent, skill or default skill.

Reviewer: ```suggestion
   Depending on the connector configuration or the decision-making capacity of the bot, the bot will transfer to a particular agent, skill, or default skill.
```

> 3. If there is a internal error or the bot service cannot be reached the connector will transfer to a default skill set up during configuration.

Reviewer: ```suggestion
3. If there is an internal error or the bot service cannot be reached the connector will transfer to a default skill set up during configuration.
```

> 3. If there is a internal error or the bot service cannot be reached the connector will transfer to a default skill set up during configuration.

Reviewer 2: ```suggestion
3. If there is an internal error or the Third-Party connector cannot reach the APIs of e-bot7, the connector will transfer to a default skill set up during configuration.
```

{: .notice}

> If you use the node 'Handover to Agent' inside of an e-bot7 flow, the conversation won't be transferred within Conversational Cloud. Instead it will be transferred within the application of e-bot7 which is not supported by this connector.

Reviewer: I think it should be at the start of the transfer action

> If you use the node 'Handover to Agent' inside of an e-bot7 flow, the conversation won't be transferred within Conversational Cloud. Instead it will be transferred within the application of e-bot7 which is not supported by this connector.

Reviewer: ```suggestion
If you use the node 'Handover to Agent' inside of an e-bot7 flow, the conversation won't be transferred within Conversational Cloud. Instead, it will be transferred within the application of e-bot7 which is not supported by this connector.
```

Action: **CONVERSATION_TRANSFER**

##### Transfer To Skill

This option transfers the conversation to the next available agent using the provided skill.

> Message in form of: `CONVERSATION_TRANSFER _skill_`

Reviewer: ```suggestion
Message in form of: `CONVERSATION_TRANSFER _skill_` 

Please note the space between skill and prefix
```

Parameters:

- `_skill_` **(Case sensitive)** skill name in Conversational Cloud.

<img class="fancyimage" style="width:450px" src="img/ThirdPartyBots/e-bot7-transfer-to-skill.png">
<figure>Figure 2.3 Configuration for transfer to skill</figure>
<br>

##### Transfer to Agent

{: .important}
This feature is depending on [permissions](https://knowledge.liveperson.com/contact-center-management-messaging-operations-transfer-to-agent.html#permissions)

> This option transfers the conversation to the particular agent matching the provided agentId and skill. If the agent is not available, the conversation will be transferred to an available agent with the same skill

Reviewer: ```suggestion
This option transfers the conversation to the particular agent matching the provided `agentId` and `skill`. If the agent is not available, the conversation will be transferred to an available agent with the same skill
```

> Message in form of: `CONVERSATION_TRANSFER _skill_ _agentId_`

Reviewer: ```suggestion
Message in form of: `CONVERSATION_TRANSFER _skill_ _agentId_` 

Please note the spaces between `skill` and `agentId` and also the order in which they are written
```

---
pagename: Basic Content
sitesection: Documents
categoryname: "Conversational AI"
documentname: Third-Party Bots
subfoldername: e-bot7
permalink: third-party-bots-e-bot7-basic-content.html
indicator:
---

### The Welcome Event

The behavior of the welcome event is different depending on whether the bot is for chat and messaging. This divergence comes down to the way that each individual LivePerson product works.

#### Messaging Conversations

A Messaging conversation qualifies as "initiated" from a Conversational Cloud perspective only after the consumer sends
their first message. The consumer is prompted for their initial message in the channel they have chosen to initiate the
conversation. As a result, the consumer’s first message will be the entry point for the main flow in e-bot7, which can be processed by a `Visitor Input` node.
In this scenario, all bot messages in the flow which are created before this node will be skipped.

> <img class="fancyimage" style="width:450px" src="img/ThirdPartyBots/e-bot7-messaging-flow.png">

Reviewer 3: @itay1313, did we give any guidance in the past around using HTML markup in Markdown files? I’d be inclined to use Markdown only, but this here is not the first time we do this. We could make an exception here for the images and figures, but should clean up later, and set up a policy for this.

Reviewer 4: good question, I agree with you

<figure>Figure 2.1 Example conversation flow for messaging without support of the `Welcome` event</figure>

> <br>

Reviewer 3: Please remove this and other (3) line breaks.

(If there is a styling issue, let @itay1313 and me know and we can advise—in the worst case we insert another line in the Markdown.)

Author: @j9t Removed all occurrences of br tags. But now there is no space between figures and their next lines. If I would add empty lines, they get killed by auto formatting which is as expected.

Problem here: figure tag doesn't refer to any visible style. I would expect to have the text inside centered (and italic) with smaller font-size with more margin to the bottom and less to the top.

<img width="978" alt="Screenshot 2022-07-04 at 09 52 59" src="https://user-images.githubusercontent.com/1105677/177109019-17812970-8b4b-4c35-8a4d-2c2f81088862.png">

Reviewer 3: Let’s do the following, and let’s do this right:

Please switch to the following structure for all figures:

```html
<figure>
 <img src="ex-a.png" alt="Two squiggles on a dirty piece of paper.">
 <figcaption>Exhibit A. The alleged <cite>rough copy</cite> comic.</figcaption>
</figure>
```

I.e., in this particular case (simplified, check on need for classes):

```html
<figure>
 <img src="img/ThirdPartyBots/e-bot7-messaging-flow.png" alt="TODO">
 <figcaption>Figure 2.1: Example conversation flow for messaging without support of the `Welcome` event.</figcaption>
</figure>
```

For styling, @itay1313, could you help @cscharun if need be so that this matches/resembles our current design?

(As we’re just getting started with a new team on the dev site, let’s make adjustments where needed. For example, if it slows us down to coordinate on the design, and if looks largely okay, let’s merge, and Itay and I will make the design adjustments separately.)

Thanks all!

Author: Hey @j9t, updated as suggested. @itay1313 Please take care of the styles. I currently left the explicit specification of the width in the style attribute to keep images sharp.

In this scenario, all bot messages in the flow which are created before this node will be skipped.

<img class="fancyimage" style="width:450px" src="img/ThirdPartyBots/e-bot7-messaging-flow.png">
<figure>Figure 2.1 Example conversation flow for messaging without support of the `Welcome` event</figure>
<br>

#### Chat Conversations

A Chat conversation is considered started when the chat is routed to an agent. The best practice is for the agent to provide the first response.
In this scenario, all bot messages in the flow which are created before the first `Visitor Input` node will be processed.
Make sure to have `Bot Message` nodes after the `Start Flow` entry point to utilize the `Welcome` event.

<img class="fancyimage" style="width:450px" src="img/ThirdPartyBots/e-bot7-chat-flow.png">
<figure>Figure 2.2 Example conversation flow for chats with support of the `Welcome` event</figure>
<br>

### Bot Messages

The flow of a bot in e-bot7 can include three different types of messages a bot can send. These are `Bot Message: Text`, `Bot Message: Image`, `Bot Message: Video`. All three types are supported with some restrictions:

> <table>

Author: @j9t Did an update here and use MD tables now. Please approve.
